scripts/azpipelines/setup-python.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
+pyroot="$AGENT_WORKFOLDER"
+
+if [ "$AGENT_OS" != "Windows_NT" ]; then
+    python_version=3.8.5
+
+    echo "Installing pyenv..."
+    export PYENV_ROOT="$AGENT_WORKFOLDER/.pyenv"
+    archive="$PYENV_ROOT/pyenv.tar.gz"
+    mkdir -p "$PYENV_ROOT"
+    curl -s -S -L "https://github.com/pyenv/pyenv/archive/v1.2.20.tar.gz" > "$archive"
+    echo "5ecd10d3ec502ce9b7d8109fbe8cb2e4a8af8b73ce5c216b8e268452724a65f3  $archive" | shasum -c
+    tar xzf "$archive" -C "$PYENV_ROOT" --strip-components=1
+    export PATH="$PYENV_ROOT/bin:$PATH"
+    pyenv --version
+
+    echo "Installing Python $python_version..."
+    env PYTHON_CONFIGURE_OPTS="--enable-shared" pyenv install $python_version
+    pyenv global 3.8.5
+    eval "$(pyenv init -)"
+
+    echo "Setting up Python virtual environment..."
+    python -m venv "$pyroot"
```

Comment:
Doesn't this create a virtual environment in the root of the git repo as opposed to inside a .venv directory?

Reply:
Also seems like this virtual environment creation/activation should happen outside of the Windows check so we can consistently use a venv for all OSes.

Reply:
`$pyroot` is set above (see like 9) to be `.venv/batchexplorer`. As per my other comment, we don't use virtualenv for Windows. I'm not sure why.

Reply:
I'm doing a text search, so I'm not sure how I'm missing it. All I see is this line:

```
pyroot="$AGENT_WORKFOLDER"
```

Reply:
Ah, just noticed it's fixed now - resolving

Reply:
It's now set (unconditionally) on [Line 5](https://github.com/Azure/BatchExplorer/blob/9f9789ed06ac2524debd4da7a6d240a00ad01d30/scripts/azpipelines/setup-python.sh#L5)